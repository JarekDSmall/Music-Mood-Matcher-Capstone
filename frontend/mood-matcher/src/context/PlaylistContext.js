import React, { createContext, useState, useContext } from 'react';

const PlaylistContext = createContext();

export const PlaylistProvider = ({ children }) => {
    const [playlists, setPlaylists] = useState([]);

    const addPlaylist = (playlist) => {
        setPlaylists(prevPlaylists => [...prevPlaylists, playlist]);
    };

    const removePlaylist = (playlistId) => {
        setPlaylists(prevPlaylists => prevPlaylists.filter(pl => pl.id !== playlistId));
    };

    return (
        <PlaylistContext.Provider value={{ playlists, setPlaylists, addPlaylist, removePlaylist }}>
            {children}
        </PlaylistContext.Provider>
    );
};

export const usePlaylist = () => {
    const context = useContext(PlaylistContext);
    if (!context) {
        throw new Error('usePlaylist must be used within a PlaylistProvider');
    }
    return context;
};
