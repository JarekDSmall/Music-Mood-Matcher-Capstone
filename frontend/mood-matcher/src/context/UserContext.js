import React, { createContext, useState, useContext } from 'react';

const UserContext = createContext();

export const UserProvider = ({ children }) => {
    const [user, setUser] = useState(null);
    const [isAuthenticated, setIsAuthenticated] = useState(false);

    return (
        <UserContext.Provider value={{ user, setUser, isAuthenticated, setIsAuthenticated }}>
            {children}
        </UserContext.Provider>
    );
};

export const useUser = () => {
    const context = useContext(UserContext);
    if (!context) {
        throw new Error('useUser must be used within a UserProvider');
    }
    return context;
};
