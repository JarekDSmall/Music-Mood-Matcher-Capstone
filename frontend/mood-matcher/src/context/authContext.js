import React, { createContext, useContext, useState } from 'react';

export const AuthContext = createContext();

export const useAuth = () => {
    return useContext(AuthContext);
};

export const AuthProvider = ({ children }) => {
    const [currentUser, setCurrentUser] = useState(null);
    const [isAuthenticated, setIsAuthenticated] = useState(false);

    const login = (user) => {
        setCurrentUser(user);
        setIsAuthenticated(true);
    };

    const logout = () => {
        setCurrentUser(null);
        setIsAuthenticated(false);
    };

    const register = async (email, password) => {
        // Here, you'll typically call your backend API to register the user.
        // For now, I'll just simulate a successful registration.
        setCurrentUser({ email }); // This is just a mock. Replace with actual user data from your backend.
        setIsAuthenticated(true);
    };

    const contextValue = {
        currentUser,
        isAuthenticated,
        login,
        logout,
        register  // <-- Added the register function here
    };

    return (
        <AuthContext.Provider value={contextValue}>
            {children}
        </AuthContext.Provider>
    );
};

