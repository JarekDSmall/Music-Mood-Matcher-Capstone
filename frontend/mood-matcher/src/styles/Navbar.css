/* Navbar.css */
@import url('https://fonts.googleapis.com/css2?family=Pacifico&display=swap');

.navbar {
    background-color: #FFDB58; /* Mustard */
    padding: 1rem 2rem;
    font-family: 'Erica One', cursive;
    border-bottom: 5px solid #996515; /* Bronze */
}

.navbar-container {
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.navbar-logo {
    color: #FFFDD0; /* Cream */
    text-decoration: none;
    font-size: 2rem;
    border: 2px solid #FFFDD0; /* Cream */
    padding: 0.2rem 0.5rem;
    border-radius: 5px;
}

.navbar-links {
    display: flex;
    gap: 1rem; /* Adjust the gap as needed */
    align-items: center; /* Ensure vertical centering */
}

.navbar-links a,
.navbar-links div,
.navbar-links button {
    color: #FFFDD0; /* Cream */
    padding: 0.5rem 1rem;
    border-radius: 5px;
    transition: background-color 0.3s, color 0.3s, text-decoration 0.3s; /* Add transition for text-decoration */
    font-size: 1.2rem;
    border: 2px solid transparent;
    background-color: transparent; /* Transparent background */
    cursor: pointer; /* Cursor pointer for button */
    font-family: 'Erica One', cursive; /* Apply the same font */
    text-decoration: none; /* No underline by default */
}

.navbar-links a:hover,
.navbar-links button:hover {
    background-color: #CC5500; /* Burnt Orange */
    color: #FFFDD0; /* Cream */
    border: 2px solid #996515; /* Bronze */
    text-decoration: underline; /* Underline on hover */
}

.active a {
    background-color: #CC5500; /* Burnt Orange */
    color: #FFFDD0; /* Cream */
    border: 2px solid #996515; /* Bronze */
}
