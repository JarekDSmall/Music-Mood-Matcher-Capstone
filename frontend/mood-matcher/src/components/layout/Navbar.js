import React from 'react';
import { Link, useNavigate } from 'react-router-dom';
import { useAuth } from '../../context/authContext';
import '../../styles/Navbar.css';

const Navbar = () => {
    const navigate = useNavigate();
    const { isAuthenticated, logoutFromSpotify } = useAuth(); // Ensure you're using the correct logout function name

    const handleLogout = async () => {
        try {
            await logoutFromSpotify(); // Use the correct logout function
            navigate('/'); // Redirect to home page after successful logout
        } catch (error) {
            console.error("Failed to log out:", error);
        }
    };

    return (
        <nav className="navbar">
            <div className="navbar-container">
                <Link to="/" className="navbar-logo">
                    MoodMatcher
                </Link>
                <div className="navbar-links">
                    {isAuthenticated ? (
                        <>
                            <div><Link to="/">Home</Link></div>
                            <div><Link to="/mood-playlist-creator">Playlist Creator</Link></div>
                            <div><Link to="/spotify">Dashboard</Link></div>
                            <div><button onClick={handleLogout}>Logout</button></div> 
                        </>
                    ) : (
                        // No content for unauthenticated users
                        null
                    )}
                </div>
            </div>
        </nav>
    );
};

export default Navbar;
