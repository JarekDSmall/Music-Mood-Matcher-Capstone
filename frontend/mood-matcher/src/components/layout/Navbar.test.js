import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from './Navbar';
import renderer from 'react-test-renderer';
import { AuthContext } from '../../context/authContext'; // Corrected import path

// Mock the auth context
const mockAuthContext = {
  isAuthenticated: false,
  logoutFromSpotify: jest.fn(),
  // Add other properties and methods if needed by Navbar component
};

describe('Navbar Component', () => {
  it('renders without crashing', () => {
    render(
      <Router>
        <AuthContext.Provider value={mockAuthContext}>
          <Navbar />
        </AuthContext.Provider>
      </Router>
    );
    // Add assertions if needed
  });

  // Snapshot Test
  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <AuthContext.Provider value={mockAuthContext}>
            <Navbar />
          </AuthContext.Provider>
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // ... other tests
});
