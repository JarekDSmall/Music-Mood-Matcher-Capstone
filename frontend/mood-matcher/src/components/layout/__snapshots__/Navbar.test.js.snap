// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component matches snapshot 1`] = `
<nav
  className="navbar"
>
  <div
    className="navbar-container"
  >
    <a
      className="navbar-logo"
      href="/"
      onClick={[Function]}
    >
      MoodMatcher
    </a>
    <div
      className="navbar-links"
    />
  </div>
</nav>
`;
